src/Components/Server/src/Circuits/RemoteRendererException.cs

```diff
@@ -5,17 +5,17 @@
 
 namespace Microsoft.AspNetCore.Components.Web.Rendering
 {
-    /// <summary>
-    /// Represents an exception related to remote rendering.
-    /// </summary>
     internal class RemoteRendererException : Exception
     {
-        /// <summary>
-        /// Constructs an instance of <see cref="RemoteRendererException"/>.
-        /// </summary>
-        /// <param name="message">The exception message.</param>
         public RemoteRendererException(string message) : base(message)
         {
         }
+
+        public RemoteRendererException(bool badInput, string message) : this (message)
+        {
+            BadInput = badInput;
+        }
+
+        public bool BadInput { get; }
```

Comment:
Can we find a better piece of terminology for this? Like `IsMisbehavingClient`? Something more descriptive

Reply:
I couldn't really think of a better name, I considered your suggestion, It might be me, but I interpret Misbehaving in this context as "Malicious" and I don't like to give it that implication.

There are different things we could do otherwise:
* Throw a more specialized exception. InvalidRenderBatchAcknowledge exception.
* Have a boolean to indicate that this was caused by an invalid acknowledge.
* Have an enum with the reasons why this failed.

In the end, all three options achieve the same goal, which is to allow us to differentiate specific situations that can be caused by bad input/bad clients from things we do care about.

Whatever option we choose I don't think it matters at all given this class is internal.
* bool is the cheapest.
* Followed by the enum.
* Having a derived class is the most expensive.

I would be ok with being explicit here and naming it InvalidRenderBatchAcknowledge or similar, thoughts?

Reply:
> Throw a more specialized exception. InvalidRenderBatchAcknowledge exception.

I'd go with that if possible as it's super clear. I'm not sure in what sense it would be more expensive.

Reply:
Ok, seems you feel super strong about having an additional exception type, so I'll just add one.

Reply:
Added an additional exception type.